src/Features/CSharp/Portable/ExtractMethod/CSharpMethodExtractor.TriviaResult.cs

```diff
@@ -32,30 +32,49 @@ private CSharpTriviaResult(SemanticDocument document, ITriviaSavedResult result)
 
             protected override AnnotationResolver GetAnnotationResolver(SyntaxNode callsite, SyntaxNode method)
             {
-                if (callsite == null || !(method is MethodDeclarationSyntax methodDefinition))
+                if (callsite == null)
                 {
                     return null;
                 }
 
-                return (node, location, annotation) => AnnotationResolver(node, location, annotation, callsite, methodDefinition);
+                if (method is MethodDeclarationSyntax methodDefinition)
+                {
+                    return (node, location, annotation) => AnnotationResolver(node, location, annotation, callsite, methodDefinition.Body, methodDefinition.ExpressionBody, methodDefinition.SemicolonToken);
+                }
+                else if (method is LocalFunctionStatementSyntax localMethodDefinition)
+                {
+                    return (node, location, annotation) => AnnotationResolver(node, location, annotation, callsite, localMethodDefinition.Body, localMethodDefinition.ExpressionBody, localMethodDefinition.SemicolonToken);
+                }
+                else
+                {
+                    return null;
+                }
             }
 
             protected override TriviaResolver GetTriviaResolver(SyntaxNode method)
             {
-                if (!(method is MethodDeclarationSyntax methodDefinition))
+                if (method is MethodDeclarationSyntax methodDefinition)
+                {
+                    return (location, tokenPair, triviaMap) => TriviaResolver(location, tokenPair, triviaMap, methodDefinition.Body, methodDefinition.ExpressionBody, methodDefinition.SemicolonToken);
+                }
+                else if (method is LocalFunctionStatementSyntax localMethodDefinition)
+                {
+                    return (location, tokenPair, triviaMap) => TriviaResolver(location, tokenPair, triviaMap, localMethodDefinition.Body, localMethodDefinition.ExpressionBody, localMethodDefinition.SemicolonToken);
```

Comment:
all you're doing in all these pieces of duplicated code is getting the body, expr-body and semicolon.  So make a single helper that deconstructs the different nodes into these thress pieces and use that.  Then, all these resolver functions don't have to care.  they can just operate on SyntaxNode.